.github/workflows/build.yml

```diff
@@ -0,0 +1,40 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+      
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+          
+      - name: Build
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: clean build
+          gradle-home-cache-cleanup: true
+
+      - name: Publish to Maven Repository
+        if: ${{ success() && github.repository == 'GeyserMC/geyser-adapters' && github.ref_name == 'master' }}
+        uses: gradle/gradle-build-action@v2
+        env:
+          ORG_GRADLE_PROJECT_opencollabRepoUsername: ${{ vars.DEPLOY_USER }}
+          ORG_GRADLE_PROJECT_opencollabRepoPassword: ${{ secrets.DEPLOY_PASS }}
```

Comment:
Named the key based off <https://docs.gradle.org/current/samples/sample_publishing_credentials.html> and the fact that i set the repo name to `opencollabRepo` in publish-conventions.